client/components/RestaurantList.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import RelatedRestaurants from './RelatedRestaurants';
+
+class RestaurantList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    };
+  }
+
+  render() {
+    const { data, cuisine, district } = this.props;
+    let list = [];
+    const type = data.filter(restaurant => restaurant.cuisine === cuisine);
```

Comment:
Good filtering system for finding related restaurants! I do think, however, it would be worth doing some of the filtering and length limiting on the server side. A criteria parameter on your find method can help filter your initial query and mongoose's '.limit()' can help limit the number of restaurants you receive from the database.

Reply:
Good idea! I'll go back and review the find filter logic for mongoose db so I can do all the logic on the database side.